go/api/base_client.go

```diff
@@ -53,15 +53,15 @@ type payload struct {
 //export successCallback
 func successCallback(channelPtr unsafe.Pointer, cResponse *C.struct_CommandResponse) {
 	response := cResponse
-	resultChannel := *(*chan payload)(channelPtr)
+	resultChannel := *(*chan payload)(getPinnedPtr(channelPtr))
 	resultChannel <- payload{value: response, error: nil}
```

Comment:
Just a question should we also the select for `Safe Handling of resultChannel`  as this can help us to catch any cases where the resultChannel is unavailable, like as follows :-  
```
select {
case resultChannel <- payload{value: response, error: nil}:
default:
    // Log the issue instead of panic
    fmt.Println("Warning: resultChannel is unavailable")
}
```

Reply:
Hi @omanges,

I think we shouldn’t. The default select case will be executed when there is no receiver on the channel, but we can’t guarantee the receiver is already waiting for it. So, doing that will randomly drop responses.